lib/trace_location/collector.rb

```diff
@@ -14,18 +13,23 @@ def collect(match:, ignore:, &block)
         hierarchy = 0
         id = 0
         cache = {}
+        method_source_cache = {}
 
         tracer = TracePoint.new(:call, :return) do |trace_point|
           next if match && !trace_point.path.to_s.match?(/#{Array(match).join('|')}/)
           next if ignore && trace_point.path.to_s.match?(/#{Array(ignore).join('|')}/)
 
           id += 1
-          caller_path, caller_lineno = trace_point.binding.of_caller(2).source_location
+          caller_loc = caller_locations(2, 1)[0]
+          caller_path = caller_loc.absolute_path
+          caller_lineno = caller_loc.lineno
           location_cache_key = "#{caller_path}:#{caller_lineno}"
 
           mes = extract_method_from(trace_point)
           next if mes.source_location[0] == '<internal:prelude>'
 
+          method_source = method_source_cache[mes] ||= remove_indent(mes.source)
```

Comment:
`UnboundMethod#source` is really heavy. So caching them improves performance dramatically.